lib/dora-parser/src/parser.rs

```diff
@@ -61,6 +61,11 @@ impl<'a> Parser<'a> {
         self.init()?;
         let mut elements = vec![];
 
+        let mut module_decl = None;
```

Comment:
I think it is better to be more forgiving and parse this as top-level-element. We need to parse the open module `module x {}` as well anyways. In one of the checkers we can then report an error when there is more than one open module definition or it's not at the top of the file.